models.js

```diff
@@ -10,3 +11,13 @@ exports.retrieveTopics = () => {
         next(err);
     })
 }
+
+exports.retrieveArticleById = (id) => {
+    return db.query('SELECT * FROM articles WHERE article_id = $1;', [id])
+    .then((queryResult) => {
+        if (!queryResult.rows[0]) {
+            return {}
```

Comment:
Instead of returning an empty object, this is where we'd want a Promise.Reject to send back an error object detailing what went wrong. In this case, that would be a non existent article. This would be a 404 error. This is what we'd call a custom error because we're having to write the details and reject the promise ourselves.

This contrasts to where psql sends us an error when we either use the wrong data type or violate a foreign key constraint (usually inputting a foreign key where there is no such key on the table it references). You've handled the psql error well here!

Check out the notes page, under the heading for custom errors for an example of this this - https://notes.northcoders.com/courses/js-back-end/error-handling